### ISSUE TITLE
ExecutorExt

### ISSUE THREAD

**Issue by MajorBreakfast** [rust-lang-nursery/futures-rs#1142]
The executor trait looks like this:

```rust
pub trait Executor {
    fn spawn_obj(
        &mut self,
        future: FutureObj<'static, ()>,
    ) -> Result<(), SpawnObjError>;

    fn status(&self) -> Result<(), SpawnErrorKind> { ... }

    // Planned
    fn spawn(
        &mut self,
        future: dyn Future<Output = ()> + Send + 'static,
    ) -> Result<(), SpawnError> { ... }
}
```

And here's how using it looks like (uses [upcoming](https://github.com/rust-lang-nursery/futures-rs/pull/1119) `boxed` combinator):
```rust
ctx.executor().spawn_obj(future.boxed().into());

// Planned
ctx.executor().spawn(future);
```

I would like to define `ExecutorExt`:
```rust
pub trait ExecutorExt {
    fn spawn_obj_with_handle<T>(
        &mut self,
        future: FutureObj<'static, T>,
    ) -> Result<JoinHandle<T>, SpawnObjError>;

    // Planned
    fn spawn_with_handle<T>(
        &mut self,
        future: dyn Future<Output = T> + Send + 'static,
    ) -> Result<JoinHandle<T>, SpawnError>;
}
```

TL;DR Move our existing `spawn_with_handle` function into `ExecutorExt`

**Comment by MajorBreakfast** [comment]
Closed by https://github.com/rust-lang-nursery/futures-rs/pull/1156
